transport_nantes/topicblog/test_tn_links.py

```diff
@@ -24,42 +24,54 @@ def test_broken_links(self):
+        self.assertEqual(self.parser.transform(
+            '[[hello]]((goodbye))'), '[[hello]]((goodbye))')
+        self.assertEqual(self.parser.transform(
+            '[[hello:goodbye'), '[[hello:goodbye')
+        self.assertEqual(self.parser.transform(
+            '[[hello:goodbye]'), '[[hello:goodbye]')
         self.assertEqual(self.parser.transform('dog [[hello:goodbye]] cat'),
                          'dog [[hello:goodbye]] cat')
 
     def test_bad_url(self):
         self.assertEqual(self.parser.transform(
             '[[action:Do something!]]((does-not-exist))'),
-                         don.action_button(reverse('topic_blog:view_item_by_slug',
-                                                   args=['does-not-exist']),
-                                           'Do something!'))
+            don.action_button(reverse('topic_blog:view_item_by_slug',
+                                      args=['does-not-exist']),
+                              'Do something!'))
 
     def test_buttons(self):
-        self.assertEqual(self.parser.transform('[[don:]]((give!))'), \
+        self.assertEqual(self.parser.transform('[[don:]]((give!))'),
                          don.bouton_don('give!'))
-        self.assertEqual(self.parser.transform('[[don:large]]((give!))'), \
+        self.assertEqual(self.parser.transform('[[don:large]]((give!))'),
                          don.bouton_don_lg('give!'))
-        self.assertEqual(self.parser.transform('[[don:adhésion]]((give!))'), \
+        self.assertEqual(self.parser.transform('[[don:adhésion]]((give!))'),
                          don.bouton_join('give!'))
         self.assertEqual(self.parser.transform(
-            '[[contact:Hello, World!]]((Je veux être bénévole))'), \
-                         don.contact_button('Hello, World!', 'Je veux être bénévole'))
-        self.assertEqual(self.parser.transform('[[don:fixed|1]]((give!))'), \
+            '[[contact:Hello, World!]]((Je veux être bénévole))'),
+            don.contact_button('Hello, World!', 'Je veux être bénévole'))
+        self.assertEqual(self.parser.transform('[[don:fixed|1]]((give!))'),
                          don.fixed_amount_donation_button(1, 'give!'))
-        self.assertEqual(self.parser.transform('[[don:fixed|5]]((give!))'), \
+        self.assertEqual(self.parser.transform('[[don:fixed|5]]((give!))'),
                          don.fixed_amount_donation_button(5, 'give!'))
 
-    ## Disabled due to captcha matching.
-    ## Think of better way to test [[news:*]]((...)).
-    #def test_news(self):
-    #    settings.csrf_token = 'some text that must exist'
-    #    self.assertEqual(self.parser.transform('[[news:kangaroo]]((aardvark))'), \
-    #                     render_inclusion_tag_to_html('newsletter',              \
-    #                                                  'show_mailing_list',       \
-    #                                                  {'mailinglist': 'kangaroo'}))
+    def test_news(self):
```

Comment:
This isn't related in any way to the task at hand, even though it's a good fix.  It should really be a separate commit (with or without issue... ;-) and with commit message about test the tn-md news tag.

Meanwhile, aside, you have added the "panel" directive but haven't tested it.

Reply:
I did actually test it, see : 
https://github.com/transport-nantes/tn_web/blob/6c5fa78caa585ce96b7b3b82e1a7073ab969b29f/transport_nantes/topicblog/templatetags/test_templatags.py#L202-L215

I didn't put it in the test_tn_links because to the difference of the other md stuff, the rendering aspect does rely on a database object witch is created in setUp. It was out of place in test_tn_links.py, in my opinion.

Reply:
Also, made issue + separate commit for the news tag, as requested :)

Reply:
Testing the markdown and testing the templatetag are two different things.  The markdown must be tested.

Reply:
> {% tn_markdown "[[panel:]]((test-panel))" %} 

tn_markdown is the tag we use to make text go through our parser, it's **not** related to tbpanels, which have their own template tag, namely {% panel 'slug-here' %}